packages/core/android/src/main/java/io/embrace/embracewrapper/EmbraceManagerModule.java

```diff
@@ -562,6 +568,110 @@ public void logNetworkClientError(String url,
 
     }
 
+    @ReactMethod()
+    public void startSpanWithName(String name, String parentSpanId, Promise promise) {
+        try{
+            promise.resolve(Embrace.getInstance().getReactNativeInternalInterface().startSpan(name, parentSpanId));
+        }catch(Exception e){
+            promise.resolve(null);
+        }
+    }
+
+    private Map<String, String> convertToReadableMap(ReadableMap readableMap) {
+        if (readableMap == null) {
+            return null;
+        }
+
+        Map<String, String> stringMap = new HashMap<>();
+
+        for (String key : readableMap.toHashMap().keySet()) {
+            stringMap.put(key, readableMap.getString(key));
+        }
+
+        return stringMap;
+    }
+
+    private ErrorCode getSpanErrorCodebyString(String errorCode){
+        switch (errorCode) {
+            case "Failure":
+                return ErrorCode.FAILURE;
+
+            case "UserAbandon":
+                return ErrorCode.USER_ABANDON;
+
+            case "Unknown":
+                return ErrorCode.UNKNOWN;
+
+            default:
+                return null;
+        }
+    }
+
+    private List<Map<String, Object>> transformListReadableMapToListMap(ReadableArray items) {
+        List<Map<String, Object>> objectMapList = new ArrayList<>();
+
+        try{
+            for (int i = 0; i < items.size(); i++) {
+                ReadableMap readableMap = items.getMap(i);
+
+                if (readableMap != null) {
+                    Map<String, Object> map = readableMap.toHashMap();
+                    // TODO Change when Android/iOS replace time in nano for ms
+                    if (map.containsKey("timestampNanos") && map.get("timestampNanos") instanceof Double) {
```

Comment:
So Android should not hit this as Android returns a Long if that field is populated?

Reply:
I dont have Long in my "types whitelist"
<img width="798" alt="image" src="https://github.com/embrace-io/embrace-react-native-sdk/assets/33431136/34bf9266-d56d-46b9-b869-d44fc107c394">
2 lines bellow I cast it to long